vivekreddynv/sonic-utilities
# [auto_ts] Enable register/de-register auto_ts config for APP Extension

#### What I did

Added the ability to dynamically add entries into AUTO_TECHSUPPORT_FEATURE when a new app extension is installed/updated

#### How I did it

Enhanced FeatureRegistry Class to also support AUTO_TECHSUPPORT_FEATURE table

#### How to verify it

- Unit Tests
- verify on live switch
```
root@r-leopard-simx-69:~# redis-cli -n 4 hgetall "AUTO_TECHSUPPORT_FEATURE|what-just-happened"
(empty array)
root@r-leopard-simx-69:~# redis-cli -n 4 hgetall "AUTO_TECHSUPPORT_FEATURE|swss"
1) "rate_limit_interval"
2) "600"
3) "state"
4) "enabled"

root@r-leopard-simx-69:~# sonic-package-manager install what-just-happened
what-just-happened is going to be installed, continue? [y/N]: y

root@r-leopard-simx-69:~# sonic-package-manager install what-just-happened
what-just-happened is going to be installed, continue? [y/N]: y

.............................
what-just-happened entry is added to AUTO_TECHSUPPORT_FEATURE table
root@r-leopard-simx-69:~# redis-cli -n 4 hgetall "AUTO_TECHSUPPORT_FEATURE|what-just-happened"
1) "rate_limit_interval"
2) "600"
3) "state"
4) "enabled"

cat /etc/sonic/config_db.json | jq -r '.AUTO_TECHSUPPORT_FEATURE'
{
....
"what-just-happened": {
    "state": "enabled",
    "rate_limit_interval": "600"
  }
}

cat /etc/sonic/init_cfg.json | jq -r '.AUTO_TECHSUPPORT_FEATURE'
{
....
"what-just-happened": {
    "state": "enabled",
    "rate_limit_interval": "600"
  }
}

root@r-leopard-simx-69:~# sonic-package-manager uninstall what-just-happened
Package what-just-happened is going to be uninstalled, continue? [y/N]: y
..................................................................


root@r-leopard-simx-69:~# redis-cli -n 4 hgetall "AUTO_TECHSUPPORT_FEATURE|what*"
(empty array)


```



#### Previous command output (if the output of a command-line utility has changed)

#### New command output (if the output of a command-line utility has changed)
